js-exercises/array-utils/arrayUtils.js

```diff
@@ -0,0 +1,57 @@
+function forEach(array, callbackFn) {
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
+      callbackFn(array[i], i, array)  
+    }
+  }
+
+  return;
+}
+
+function map(array, callbackFn) {
+  let resultingArray = [];
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
```

Comment:
Due to this check, `if(array[i])`, different behaviour than Array.prototype.map
`map(['1', null, '3', 4], (element, index) => typeof element === "number")`
> (3) [false, false, true]

`['1', null, '3', 4].map((element, index) => typeof element === "number")`
> (4) [false, false, false, true]